flashlight/nn/modules/Module.h

```diff
@@ -163,6 +169,23 @@ class UnaryModule : public Module {
   FL_SAVE_LOAD_WITH_BASE(Module)
 };
 
+/**
+ * Precision Cast Module. Casts the input from its original precision to the
+ * target precision. Precision cast only alters the underlying array and leaves
+ * other attributes of the input variable unchanged.
+ */
+
+class PrecisionCast : public UnaryModule {
```

Comment:
Why is precision cast a unary module? Shouldn't it be able to operate on a vector of `fl::Variable` as well and cast each element?

Reply:
True. Fixed.